packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/form/FormIdProvider.java

```diff
@@ -36,12 +37,33 @@ public FormIdProvider(IObjectService objectService) {
 
     @Override
     public String getId(FormDescription formDescription) {
-        String sourceId = this.getSourceIdFromElementDescription(formDescription);
+        String sourceIdPart = this.getSourceIdPart(formDescription);
         String sourceElementId = this.objectService.getId(formDescription);
-        return FORM_DESCRIPTION_KIND + "&" + SOURCE_KIND + "=" + VIEW_SOURCE_KIND + "&" + SOURCE_ID + "=" + sourceId + "&" + SOURCE_ELEMENT_ID + "=" + sourceElementId;
+        return FORM_DESCRIPTION_KIND + "&" + SOURCE_KIND + "=" + VIEW_SOURCE_KIND + sourceIdPart + "&" + SOURCE_ELEMENT_ID + "=" + sourceElementId;
     }
 
-    private String getSourceIdFromElementDescription(EObject elementDescription) {
-        return elementDescription.eResource().getURI().toString().split("///")[1];
+    private String getSourceIdPart(EObject description) {
+        return this.getSourceIdFromElementDescription(description).map(resourceId -> "&" + SOURCE_ID + "=" + resourceId).orElse("");
+    }
+
+    private Optional<String> getSourceIdFromElementDescription(EObject elementDescription) {
+        return Optional.ofNullable(elementDescription.eResource()).map(eResource -> {
+            String[] parts = eResource.getURI().toString().split("///");
+            if (parts.length > 0) {
+                return parts[1];
+            } else {
+                return null;
+            }
+
+        });
+
+    }
```

Comment:
view models are expected to be in a resource (and we could even say in a resource set) especially with the default colors